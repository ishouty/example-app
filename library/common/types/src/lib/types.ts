export function helloTypes(): string {
  return 'types';
}
