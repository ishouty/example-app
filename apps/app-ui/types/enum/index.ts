export * from './navigation-layout-menu.enum';
