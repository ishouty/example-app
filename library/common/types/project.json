{
  "name": "types",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "library/common/types/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/library/common/types",
        "main": "library/common/types/src/index.ts",
        "tsConfig": "library/common/types/tsconfig.lib.json",
        "assets": ["library/common/types/*.md"]
      }
    }
  }
}
