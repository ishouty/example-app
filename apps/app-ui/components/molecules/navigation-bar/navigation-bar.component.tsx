'use-client';

import classNames from 'classnames';
import { ThemeContext } from '../../../providers/theme/theme.context';
import React, { ReactNode, useContext, useEffect } from 'react';
import { FunctionComponent } from 'react';
import Link from 'next/link';
import { IconBrandGitlab, icons, IconUser } from '@tabler/icons-react';
import { NavigationLayoutMenuType } from '../../../types/enum';
import { useAppSelector } from '../../../redux/store';
import { TNavigationBarProps } from './navigation-bar.type';

export const NavigationBar: FunctionComponent<TNavigationBarProps> = ({
  items,
}) => {
  //const theme = useContext(ThemeContext);
  //const userProfile = useAppSelector((state) => state.user);

  return (
    <nav
      className={classNames(
        'bg-gray-100 flex justify-between items-center p-4 border-b border-gray-200 shadow-sm',
      )}
    >
      <div className="flex space-x-4 underline">
        {items
          .filter((item) => item.type === NavigationLayoutMenuType.main)
          .map((item, index) => {
            const Icon = item.icon;
            return (
              <Link
                href={item.link}
                key={index}
                passHref
                className="flex space-x-1"
              >
                <div className="flex items-center space-x-2">
                  {Icon && <Icon className="h-5 w-5" />}{' '}
                  <span>{item.label}</span>{' '}
                </div>
              </Link>
            );
          })}
      </div>
      <div className="flex">
        {items
          .filter((item) => item.type === NavigationLayoutMenuType.settings)
          .map((item, index) => {
            const Icon = item.icon;

            return (
              <Link href={item.link} key={index} passHref>
                <div className="flex items-center space-x-2">
                  {Icon && <Icon className="h-5 w-5" />}
                  <span>{item.label}</span>{' '}
                </div>
              </Link>
            );
          })}
      </div>
    </nav>
  );
};
