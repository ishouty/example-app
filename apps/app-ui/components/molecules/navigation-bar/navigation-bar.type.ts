import { NavigationLayoutMenuType } from 'types';

export type TNavigationBarProps = {
  items: {
    link: string;
    label: string;
    selected: boolean;
    type: NavigationLayoutMenuType;
    hideMenu: boolean;
    icon: React.ElementType;
  }[];
};
