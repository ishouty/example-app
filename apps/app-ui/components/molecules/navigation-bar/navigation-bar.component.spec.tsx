import { render, screen } from '@testing-library/react';
import { NavigationBar } from './navigation-bar.component';
import { TNavigationBarProps } from './navigation-bar.type';
import { NavigationLayoutMenuType } from 'types';
import { IconUsb } from '@tabler/icons-react';
import { text } from 'stream/consumers';

describe('Navigation Bar Component', () => {
  const setup = () => {
    const navigationItems: TNavigationBarProps = {
      items: [
        {
          link: 'http://www.google.com',
          label: 'example-link',
          selected: true,
          type: NavigationLayoutMenuType.main,
          hideMenu: false,
          icon: IconUsb,
        },
      ],
    };

    return render(<NavigationBar items={navigationItems.items} />);
  };

  test('react component matches snapshot', () => {
    const component = setup();
    expect(component.container).toMatchSnapshot();
  });

  test('renders component with navigation items a links', () => {
    const component = setup();
    expect(component.container.querySelectorAll('a')).toHaveLength(1);
    expect(screen.getByText(/example-link/i).innerHTML).toEqual('example-link');
    expect(component.container.querySelector('a')?.outerHTML).toMatch(
      /http:\/\/\www.google.com/,
    );
    expect(component.container.querySelectorAll('svg')).toHaveLength(1);
  });
});
