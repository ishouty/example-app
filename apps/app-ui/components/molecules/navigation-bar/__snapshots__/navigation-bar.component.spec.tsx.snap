// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Component react component matches snapshot 1`] = `
<div>
  <nav
    class="bg-gray-100 flex justify-between items-center p-4 border-b border-gray-200 shadow-sm"
  >
    <div
      class="flex space-x-4 underline"
    >
      <a
        class="flex space-x-1"
        href="http://www.google.com"
      >
        <div
          class="flex items-center space-x-2"
        >
          <svg
            class="tabler-icon tabler-icon-usb h-5 w-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
            />
            <path
              d="M12 17v-11.5"
            />
            <path
              d="M7 10v3l5 3"
            />
            <path
              d="M12 14.5l5 -2v-2.5"
            />
            <path
              d="M16 10h2v-2h-2z"
            />
            <path
              d="M7 9m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0"
            />
            <path
              d="M10 5.5h4l-2 -2.5z"
            />
          </svg>
           
          <span>
            example-link
          </span>
           
        </div>
      </a>
    </div>
    <div
      class="flex"
    />
  </nav>
</div>
`;
